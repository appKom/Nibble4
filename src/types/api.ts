export interface UserResponse {
  count: number;
  next: any;
  previous: any;
  results: Array<{
    pk: number;
    first_name: string;
    last_name: string;
    saldo: number;
  }>;
}

export const isUserResponse = (value: any): value is UserResponse => {
  return typeof value.count === "number" || value.results === "object";
};
