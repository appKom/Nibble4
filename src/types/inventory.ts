export type Product = {
  pk: number;
  name: string;
  price: number;
  image: {
    sm: string;
  };
  category: {
    pk: number;
    name: string;
  };
};
