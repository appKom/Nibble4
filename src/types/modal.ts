export enum modalTypes {
  PURCHASE = "PURCHASE",
  COMPLETE = "COMPLETE",
  ERROR = "ERROR",
  DISABLED = "DISABLED",
  INACTIVE = "INACTIVE",
}
