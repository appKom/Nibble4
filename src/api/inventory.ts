import { Product } from "types/inventory";
import { get } from "./index";

const fetchInventory = async (url: string): Promise<Product[]> => {
  const response = await get({ url }).catch(() => Promise.resolve(undefined));
  if (response && response.ok) {
    const json = await response.json();
    return json.sort((a: Product, b: Product) => (a.name > b.name ? 1 : -1));
  }
  return [];
};

export default fetchInventory;
