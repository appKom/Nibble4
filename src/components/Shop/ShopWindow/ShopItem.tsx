import React, { FC } from "react";
import styled from "styled-components";
import { Product } from "types/inventory";
import { IMAGE_URI } from "../../../api";

type ShopWindowItemProps = {
  storeObject: Product;
};

const ShopItem: FC<ShopWindowItemProps> = (storeObject) => {
  const imageSrc = storeObject.storeObject.image
    ? IMAGE_URI(storeObject.storeObject.image.sm)
    : "";

  return (
    <Container>
      <img src={imageSrc} alt="Product image" />

      <TextContainer>
        <span>
          <b> {storeObject.storeObject.name} </b>
        </span>

        <span>
          <b>{storeObject.storeObject.price}</b>kr
        </span>

        <DescSpan>{storeObject.storeObject.description} </DescSpan>
      </TextContainer>
    </Container>
  );
};

export default ShopItem;

const Container = styled.div`

  padding: 5px;
  background-color: white;
  border-radius: 3px;
  width: 90%;
  height: 90%;

  margin-left: auto;
  margin-right: auto;
  margin-top: 10px;
  margin-bottom: 10px;
  box-shadow: 2px 2px 7px #888888;

  display: grid;
  grid-template-rows; 4fr 3fr;

  img {
    display: block;
    margin-left: auto;
    margin-right: auto;
    width: 35%;
  }
`;

const TextContainer = styled.div`
  display: grid;
  grid-template-rows: 1fr 1fr;
  grid-template-columns: 4fr 1fr;
`;

const DescSpan = styled.span`
  color: gray;
  font-size: 12px;
`;
