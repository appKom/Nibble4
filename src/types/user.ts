import { Product } from "./inventory";

export type User = {
  pk: number;
  balance: number;
  first_name: string;
  favourites: Product[];
};
