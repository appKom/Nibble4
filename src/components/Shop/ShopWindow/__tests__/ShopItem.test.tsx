import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import ShopItem from "../ShopItem";
import { Product } from "types/inventory";

const mockProduct: Product = {
  pk: 95,
  name: "Powerking",
  description: "Energidrikk",
  price: 10,
  image: {
    sm: "/media/images/responsive/sm/38688f77-21b6-4936-8ba6-c976538fb110.png",
  },
  category: {
    pk: 3,
    name: "Drikke",
  },
};

afterEach(cleanup);
it("renders", () => {
  const { asFragment } = render(<ShopItem product={mockProduct} />);
  expect(asFragment()).toMatchSnapshot();
});
