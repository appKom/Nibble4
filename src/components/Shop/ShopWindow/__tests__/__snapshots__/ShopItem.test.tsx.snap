// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ ixLtVV"
  >
    <img
      alt="Product image"
      src="https://online.ntnu.no//media/images/responsive/sm/38688f77-21b6-4936-8ba6-c976538fb110.png"
    />
    <div
      class="sc-gsTCUz kFHCOk"
    >
      <span>
        <b>
           Powerking 
        </b>
      </span>
      <span>
        <b>
          10
        </b>
        kr
      </span>
    </div>
  </div>
</DocumentFragment>
`;
